cdrs-tokio/src/cluster/topology.rs

```diff
@@ -0,0 +1,45 @@
+use crate::cluster::connection_manager::ConnectionManager;
+use crate::load_balancing::LoadBalancingStrategy;
+use crate::transport::CdrsTransport;
+use std::marker::PhantomData;
+
+/// Allows to read current topology info from the cluster
+pub struct TopologyReader<
+    T: CdrsTransport + Send + Sync + 'static,
+    CM: ConnectionManager<T> + Send + Sync,
+    LB: LoadBalancingStrategy<T, CM> + Send + Sync,
+> {
+    //control_connection: &'a ControlConnection<T, CM, LB>,
```

Comment:
We can use tokio::mpsc to avoid directly depending on the control connection. The connection can wait for refresh events and send frames when requested. This way we also can get rid of all these generic parameters.

Reply:
Doing so creates a cyclic dependency. `TopologyReader` needs the `ControlConnection`, `ClusterMetadataManager` needs `TopologyReader` and `ControlConnection` needs `ClusterMetadataManager`.

Reply:
I might have an idea how to approach this. Let me check the datastax driver and add a reader skeleton.